sys/usb/usbus/cdc/acm/cdc_acm.c

```diff
@@ -0,0 +1,334 @@
+                            usb_setup_t *setup)
+{
+    (void)state;
+    usbus_cdcacm_device_t *cdcacm = (usbus_cdcacm_device_t*)handler;
+    switch(setup->request) {
+        case USB_CDC_MGNT_REQUEST_SET_LINE_CODING:
+            if ((state == USBUS_CONTROL_REQUEST_STATE_OUTDATA) &&
+                    (setup->length == sizeof(usb_req_cdcacm_coding_t))) {
+                size_t len = 0;
+                usb_req_cdcacm_coding_t *coding =
+                    (usb_req_cdcacm_coding_t*)usbus_control_get_out_data(usbus,
+                                                                         &len);
+
+                if (len != sizeof(usb_req_cdcacm_coding_t)) {
+                    DEBUG("CDCACM: incorrect length of LINE_CODING set request"
+                          ", expected: %u, got: %u",
+                          sizeof(usb_req_cdcacm_coding_t), len);
+                    return -1;
+                }
+                if (cdcacm->coding_cb) {
+                    DEBUG("Setting line coding to baud rate %" PRIu32 ", "
+                          "%u data bits, parity value %u, stop bit value %u\n",
+                          coding->baud, coding->databits, coding->parity,
+                          coding->format);
+                    if (cdcacm->coding_cb(cdcacm, coding->baud,
+                                          coding->databits, coding->parity,
+                                          coding->format) < 0) {
+                        return -1;
+                    }
+                }
+                memcpy(&cdcacm->coding, coding,
+                       sizeof(usb_req_cdcacm_coding_t));
+            }
+            break;
+        case USB_CDC_MGNT_REQUEST_GET_LINE_CODING:
+            usbus_control_slicer_put_bytes(usbus, (uint8_t*)&cdcacm->coding,
+                               sizeof(usb_req_cdcacm_coding_t));
+            break;
+        case USB_CDC_MGNT_REQUEST_SET_CONTROL_LINE_STATE:
+            if (setup->value & USB_CDC_ACM_CONTROL_LINE_DTE) {
+                DEBUG("CDC ACM: DTE enabled on interface %u\n", setup->index);
+                cdcacm->state = USBUS_CDCACM_LINE_STATE_DTE;
+                usbus_cdc_acm_flush(cdcacm);
+            }
+            else {
+                cdcacm->state = USBUS_CDCACM_LINE_STATE_DISCONNECTED;
+                DEBUG("CDC ACM: DTE disabled on interface %u\n", setup->index);
+            }
+            break;
+        default:
+            DEBUG("unhandled USB setup request:0x%x\n", setup->request);
+            return -1;
+    }
+    return 1;
+}
+
+static void _handle_in(usbus_cdcacm_device_t *cdcacm,
+                       usbdev_ep_t *ep)
+{
+    if ((cdcacm->usbus->state != USBUS_STATE_CONFIGURED) ||
+        (cdcacm->state != USBUS_CDCACM_LINE_STATE_DTE)) {
+        return;
+    }
+    while (!tsrb_empty(&cdcacm->tsrb)) {
+        int c = tsrb_get_one(&cdcacm->tsrb);
+        ep->buf[cdcacm->occupied++] = (uint8_t)c;
+        if (cdcacm->occupied >= 64) {
```

Comment:
this should better be a `#define` or a `sizeof`

Reply:
Fixed